WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift

```diff
@@ -1458,6 +1460,34 @@ private extension ProductFormViewController {
     }
 }
 
+// MARK: Action - Show Custom Fields
+private extension ProductFormViewController {
+    func showCustomFields() {
+        guard let product = product as? EditableProductModel else {
+            return
+        }
+
+        let customFields = product.product.customFields.map {
+            CustomFieldViewModel(metadata: $0)
+        }
+
+        let customFieldsView = UIHostingController(
+            rootView: CustomFieldsListView(
+                isEditable: true,
+                viewModel: CustomFieldsListViewModel(customFields: customFields),
+                onBackButtonTapped: { [weak self] in
+                    // Restore the hidden navigation bar
+                    self?.navigationController?.setNavigationBarHidden(false, animated: true)
+            })
+        )
+
+        // Hide the navigation bar as `CustomFieldsListView` will create its own toolbar.
+        navigationController?.setNavigationBarHidden(true, animated: true)
```

Comment:
The animation causes unwanted jiggling between transitions, so setting `animated` to `false` might help.

For a smoother experience, please consider adding a separate type `CustomFieldsListHostingController` and set the navigation items for the Save and Add buttons in that controller if it's editable. That way you can reuse the back button and avoid having a thinner chevron icon on this screen compared to the product form screen.

Reply:
Thanks, Paolo also raised something similar here https://github.com/woocommerce/woocommerce-ios/pull/14059#pullrequestreview-2337523657 and I'll quote my reply:

> I agree with that, though, and in hindsight it would've been more consistent to copy Blaze's method of creating navigation coordinator (`BlazeCampaignCreationCoordinator`) and use UIKit navigation still. While I do find that the experience of setting up navigation using SwiftUI's `NavigationStack` to be more intuitive and concise, it's also unfortunate that connecting it with UIKit's navigation isn't as smooth as possible.

> An improvement here would be to refactor the navigation to use coordinator. Given what's already set up, this would be a decently sized task, so let's keep this for now. In the meanwhile I'll add a bit more things like turning off animation and making the back icon bolder, to make the back navigation look less odd.

Reply:
I added https://github.com/woocommerce/woocommerce-ios/pull/14068/commits/cfd6c47bbbb2e53ec6be1be69c20be0ece84e24f to:
- remove animations
- make the back button thicker (good catch on this! This is also done in other parts of the code that uses the chevron)

Reply:
Something I'm not able to fix is that the placement of the back arrow has extra padding in the list view, compared to the back button on the product details or custom fields editor view:

https://github.com/user-attachments/assets/04aa0068-970c-4301-8e4a-04d96912b7ad

I tried a few things:
- adding negative padding: this works but make the target area smaller and taps don't register on the left side
- using `.topBarLeading` instead of `.cancellationAction` as placement
- using `ToolbarItemGroup`

But none of them fixes the problem. It's odd because the back button generated automatically by SwiftUI (in the editor view) has the correct placement similar to UIKit's back button in the product details screen. However, the back button created with `Toolbar` has a different leading padding. This feels like a SwiftUI bug to me, and as it's a bit odd but not breaking functionality, I think we can keep it like that for now.